Source/Cake.Recipe/Content/buildProvider.cake

```diff
@@ -72,7 +72,7 @@ public enum BuildProviderType
 
 public static IBuildProvider GetBuildProvider(ICakeContext context, BuildSystem buildSystem)
 {
-    if (buildSystem.IsRunningOnAzurePipelines || buildSystem.IsRunningOnAzurePipelinesHosted)
+    if (buildSystem.IsRunningOnAzurePipelines || buildSystem.AzurePipelines.Environment.Agent.IsHosted)
```

Comment:
If you want to check if you run on Azure Pipelines (On-Prem or hosted) checking for `buildSystem.IsRunningOnAzurePipelines` should be enough.

Reply:
Sounds like a plan! Thank you!